gateway/src/apicast/upstream.lua

```diff
@@ -144,6 +144,10 @@ local function host_header(uri)
     end
 end
 
+function _M:set_host_for_host_header(host)
```

Comment:
Aren't we just using `host = ...` on other places? I mean without a method, just setting the property.

Reply:
I was doubting between adding this new method or not.

In the end, I added it because I think that it makes the code easier to understand. Using the method it's pretty clear that I'm only using the host to set the Host header. However, with `.host = ...`, it's not that clear, the host could be used for other things.

Reply:
Agreed. Maybe we could tune up the naming a bit? `use_host_header` ?
And use it on other places where `.host = `  is used right now?

Reply:
Sounds good :+1:

Reply:
Renamed in https://github.com/3scale/apicast/pull/983/commits/4b898cfb13ca98d94697451547e6d3e49f976b3d

Modified calls to the new method from `.host =` in https://github.com/3scale/apicast/pull/983/commits/a399634b5b3f88237f92ffc158cf8fe50518fc01